packages/jest-mock/src/index.ts

```diff
@@ -1089,12 +1089,11 @@ export class ModuleMocker {
     methodKey: K,
   ): V extends ClassLike | FunctionLike ? Spied<V> : never;
 
-  // eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
-  spyOn<T extends object, K extends PropertyLikeKeys<T>>(
+  spyOn<T extends object>(
```

Comment:
The overloads are now way more readable. Nice to look at them.

One more detail I am leaving for the future iterations. The return type of `spyOn` and `_spyOnProperty` methods is `MockInstance`, but the value wich is return has to be cast using `as Mock`. This is because of the return type of `_makeComponent` method. It comes from overloads which are causing some type issues. That needs more thinking. So I am leaving this for the future.